src/Query/Processor.php

```diff
@@ -103,13 +103,21 @@ public function processColumns($results)
 
     /**
      * {@inheritDoc}
-     * @param array{ index_name: string }&array<string, mixed> $results
-     * @return array<array-key, string>
+     * @param list<array<string, mixed>> $results
+     * @return list<array{name: string, columns: list<string>, type: string, unique: bool, primary: bool}>
      */
     public function processIndexes($results)
     {
         return array_map(function ($result) {
-            return ((object) $result)->index_name;
+            $result = (object) $result;
+
+            return [
+                'name' => $name = $result->name,
+                'columns' => $result->columns ? explode(',', $result->columns) : [],
+                'type' => strtolower($result->type),
+                'unique' => (bool) $result->unique,
+                'primary' => $name === 'PRIMARY_KEY',
+            ];
```

Comment:
memo: Shape similar to https://github.com/laravel/framework/blob/12.x/src/Illuminate/Database/Query/Processors/MySqlProcessor.php#L86